code/datums/noobmode.dm

```diff
@@ -0,0 +1,113 @@
+/datum/action/equipHazard
+	name = "Equip skinsuit"
+	desc = "hazard gear and internals. Takes a few seconds."
+	icon_icon = 'icons/obj/clothing/suits.dmi'
+	button_icon_state = "skinsuit"
+	var/obj/item/clothing/suit/space/skinsuit/suitslot
+	var/obj/item/clothing/mask/breath/maskslot
+	var/obj/item/clothing/head/helmet/space/helmetslot
+	var/obj/item/tank/internals/airslot
+	var/obj/item/storage/box/survBox
+	var/paniced = FALSE
+
+/datum/action/equipHazard/Trigger()
+	if(ishuman(owner))
+		var/mob/living/carbon/human/H = owner
+		testing(paniced)
+		if(!paniced)
+			if(panicEquip(H))
+				toggle()
+		else
+			if(stowEquipment(H))
+				toggle()
+
+
+/datum/action/equipHazard/proc/GatherItems(obj/item/storage/CNT)
+	var/stuff = CNT.contents
+	airslot = locate(/obj/item/tank/internals) in stuff
+	helmetslot = locate(/obj/item/clothing/head/helmet/space) in stuff
+	maskslot = locate(/obj/item/clothing/mask/breath) in stuff
+	suitslot = locate(/obj/item/clothing/suit/space/skinsuit) in stuff
+	if(!airslot || !helmetslot || !maskslot || !suitslot)
+		return FALSE
+	return TRUE
+
+/datum/action/equipHazard/proc/toggle()
+	if(paniced)
+		paniced = FALSE
+	else
+		paniced = TRUE
+
+/datum/action/equipHazard/Grant(mob/user, obj/item/containBox)
+	. = ..()
+	survBox = containBox
+
+/datum/action/equipHazard/proc/stowEquipment(mob/living/carbon/human/USR,speed = 10)
+	to_chat(USR,"<span class='notice'> You stuff the emergency equipment back into the box")
+	if(!do_after(USR, speed))
+		return FALSE
+	testing("time passed")
+	if(helmetslot && helmetslot == USR.head)
+		survBox.attackby(helmetslot,USR)
+	if(maskslot && maskslot == USR.wear_mask)
+		survBox.attackby(maskslot,USR)
+	if(suitslot && suitslot == USR.wear_suit)
+		if(!suitslot.rolled_up)
+			suitslot.attack_self(USR)
+		survBox.attackby(suitslot,USR)
+	if(airslot && airslot.loc != survBox)//instead of checking every pocket we just check the turf
+		if(airslot in range(1,USR))
+			survBox.attackby(airslot,USR)
```

Comment:
Couldn't you just use `if(gatheritems(survBox))` over all of these args to see if there are any null slots, instead of individually checking every slot?

Reply:
`gatheritems` finds items in the box, which means it clears the old ones and thus now allow the deEquip

Reply:
It was the original reason I made `gatheritems` return a TRUE or false though